.github/workflows/CD.yml

```diff
@@ -0,0 +1,48 @@
+name: CD
+
+on:
+  pull_request:
+    types: [closed]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.merged == true || github.event_name == 'workflow_dispatch'
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.11
+
+    - name: Setup development environment
+      run: |
+        python -m venv .venv
+        source .venv/bin/activate
+        python -m pip install -U pip
+        pip install azdev
+        azdev setup -r .
+        azdev extension add apic-extension
+
+    # Append commit hash to the version. E.g. 1.0.0b4+abc1234
+    - name: Generate and update version
+      run: |
+        python ./github/workflows/update_version.py
+
+    - name: Build binary
+      run: |
+        source .venv/bin/activate
+        azdev extension build apic-extension
+
+    - name: Upload release asset
+      uses: actions/upload-release-asset@v1
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
you may add permission for GITHUB_TOKEN, like https://github.com/OfficeDev/TeamsFx/blob/cb0576ec8d60220cfcf6b6b96f4412eb0472a377/.github/workflows/cd.yml#L25

Reply:
you may also consider actions/upload-artifact@v3
https://github.com/OfficeDev/TeamsFx/blob/cb0576ec8d60220cfcf6b6b96f4412eb0472a377/.github/workflows/cd.yml#L283